Shared/Samples/Query feature count and extent/QueryFeatureCountAndExtentView.swift

```diff
@@ -0,0 +1,178 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct QueryFeatureCountAndExtentView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// The currently selected state abbreviation.
+    @State private var selectedState: String?
+    
+    /// A Boolean value indicating whether the feature count bar is showing.
+    @State private var showFeatureCountBar = false
+    
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    /// The current viewpoint of the map.
+    @State var viewpoint: Viewpoint?
+    
+    /// The count of features within the current viewpoint.
+    @State var featureCountResult: Int?
+    
+    var body: some View {
+        MapViewReader { proxy in
+            MapView(map: model.map)
+                .onViewpointChanged(kind: .boundingGeometry) { newViewpoint in
+                    // Update viewpoint when it changes.
+                    viewpoint = newViewpoint
+                }
+                .task(id: selectedState) {
+                    // Perform query and update the viewpoint when the selected state changes.
+                    guard let state = selectedState else { return }
+                    do {
+                        if let combinedExtent = try await model.queryExtent(stateAbbreviation: state) {
+                            await proxy.setViewpointGeometry(combinedExtent)
+                            showFeatureCountBar = false
+                        }
+                    } catch {
+                        self.error = error
+                    }
```

Comment:
```suggestion
                    }
                    selectedState = nil
```

This allows the same state to be selected twice in a row.